modules/caddyhttp/caddyhttp.go

```diff
@@ -124,6 +129,8 @@ func (app *App) Provision(ctx caddy.Context) error {
 
 	repl := caddy.NewReplacer()
 
+	certmagic.Default.DefaultServerName = app.DefaultSNI
```

Comment:
This might be racey, esp. with regard to config reloads, since this variable is global, but app.DefaultSNI is not. I am looking at a different way to set DefaultServerName on the individual certmagic configs that are made [around here](https://github.com/caddyserver/caddy/blob/dd103a67876bd19afd000bda4410fbf83ee5b57d/modules/caddytls/connpolicy.go#L165).

Reply:
I did wonder about that, in some options I went down I did set this directly at this point.

Reply:
Setting the DefaultServerName inside getCertificate would be very explicit